tests/unit/amo/components/TestAddon.js

```diff
@@ -1566,4 +1568,71 @@ describe(__filename, () => {
       expect(installButton).toHaveProp('uninstall', uninstall);
     });
   });
+  // Non-public add-ons require an account listed as a developer of the add-on or admin rights.
+  it('displays a notice to the admin on a disabled addon', () => {
+    const addon = createInternalAddon({
+      ...fakeAddon,
+      status: 'disabled',
+    });
+
+    const root = shallowRender({
+      addon,
+      store: dispatchSignInActions({
+        userProps: {
+          permissions: [ALL_SUPER_POWERS],
```

Comment:
so @willdurand this is a sufficient test? if we do `permission: []` here, the test still passes (same with the below case - if we say `is_addon_developer: false` for the dev case). I understand the API will return a 403/404 if they are not authorized/logged in but I guess I am still a little confused myself on the usefulness of these tests shown in this way b/c if you change the details in the way I noted above it still passes. Maybe we should add more comment about this not even being reached (or the test case that checks this)? If that makes sense..

But if you're ok with this, then I'll defer to you : )

Reply:
right, maybe a single test case would be enough.

Reply:
@willdurand thanks for the quick reply , should i delete the test case for developer?

Reply:
yeah, I think @rebmullin is right. we should have test cases that fail if we change something, so I suppose we only need one test case that is the opposite of when api returns 403.

Reply:
@willdurand, so there is already this test case:

```javascript
it('renders NotFound page for forbidden add-on - 403 error', () => {
    const { store } = dispatchClientMetadata();

    const error = createApiError({
      response: { status: 403 },
      apiURL: 'https://some/api/endpoint',
      jsonResponse: { message: 'You do not have permission.' },
    });
    const errorHandler = new ErrorHandler({
      id: 'some-id',
      dispatch: store.dispatch,
    });
    errorHandler.handle(error);

    const root = renderComponent({ errorHandler, store });
    expect(root.find(NotFound)).toHaveLength(1);
  });
```

is this sufficient? I mean it shows that a 403 gives Notfound page..

and maybe we can just add the following to the above test:
`expect(root.find(Notice)).toHaveLength(0);` 

would that work?

(on top of merging the above dev and admin tests as you suggest above)

Reply:
still using the comment above the test case:
`// Non-public add-ons require an account listed as a developer of the add-on or admin rights.`

Reply:
We keep the comment and we add a test case indicating that we display a notice message when an admin or the developer views the page.